dev/functional/cxx_scope_guard.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+namespace sqlite_orm::internal {
+    /*  
+        Poor-man's scope (exit) guard until C++29 finally comes with proper standard facilities [Draft D3610].
+     */
+    template<class F>
+    struct scope_guard {
+        explicit scope_guard(F f) : f{std::move(f)} {}
+        ~scope_guard() {
+            f();
+        }
+
+        F f;
```

Comment:
nit: I'd skip single character names. I'd name it `function` to make it more clear for future readers. Short variable names used to be popular when we had only 80 character limited lines on our screens. Now we don't need to the code to be encoded, it can be written as a regular text to make it easier to read it